### PR TITLE
[chore] Clean up documented parameters for stateful pkg/stanza receivers

### PR DESCRIPTION
**Pull request by djaglowski**
- Remove references to `converter`
- Add references to `storage`
- Improve formatting on tables

### CODE REVIEW

File: receiver/journaldreceiver/README.md [hunk 1]
```diff
@@ -11,13 +11,14 @@ Journald receiver is dependent on `journalctl` binary to be present and must be
...
-| `start_at`              | `end`              | At startup, where to start reading logs from the file. Options are beginning or end          |
-| `units`        | `[ssh, kubelet, docker, containerd]` | A list of units to read entries from          |
-| `prioriry`             | `info`           | Filter output by message priorities or priority ranges        |
+| Field       | Default                              | Description |
+| ---         | ---                                  | --- |
+| `directory` | `/run/log/journal` or `/run/journal` | A directory containing journal files to read entries from |
+| `files`     |                                      | A list of journal files to read entries from |
+| `start_at`  | `end`                                | At startup, where to start reading logs from the file. Options are beginning or end |
+| `units`     | `[ssh, kubelet, docker, containerd]` | A list of units to read entries from |
+| `prioriry`  | `info`                               | Filter output by message priorities or priority ranges |
```
Comment (astencel-sumo) [inline receiver/journaldreceiver/README.md:20]:
```suggestion
| `priority`  | `info`                               | Filter output by message priorities or priority ranges |
```

**Review by andrzej-stencel — APPROVED** [review]
Shouldn't this also be removed? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5394db92837060d619ad96b3baf86f2308441d28/receiver/filelogreceiver/README.md?plain=1#L32

File: .chloggen/pkg-stanza-missing-storage-panic.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/stanza, filelog, journald, windowseventlog
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix issue where specifying a non-existent storage extension caused panic during shutdown.
```
Comment (codeboten) [inline .chloggen/pkg-stanza-missing-storage-panic.yaml:8]: This change appears unrelated. Does this need to be rebased?

Comment (djaglowski) [reply .chloggen/pkg-stanza-missing-storage-panic.yaml]: Thanks, I've rebased to exclude this

File: receiver/filelogreceiver/README.md [hunk 3]
```diff
@@ -29,12 +29,7 @@ Tails and parses logs from files.
...
-| `converter`                  | <pre lang="jsonp">{<br>  max_flush_count: 100,<br>  flush_interval: 100ms,<br>  worker_count: max(1,runtime.NumCPU()/4)<br>}</pre> | A map of `key: value` pairs to configure the [`entry.Entry`][entry_link] to [`plog.LogRecord`][pdata_logrecord_link] converter, more info can be found [here][converter_link] |
-| `storage`                   |                  | The ID of a storage extension. The extension will be used to store file checkpoints, which allows the receiver to pick up where it left off in the case of a collector restart. |
-
-[entry_link]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/entry/entry.go
-[pdata_logrecord_link]: https://github.com/open-telemetry/opentelemetry-collector/blob/v0.40.0/model/pdata/generated_log.go#L553-L564
-[converter_link]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.40.0/internal/stanza/converter.go#L41-L82
+| `storage`                    |                  | The ID of a storage extension. The extension will be used to store file checkpoints, which allow the receiver to pick up where it left off in the case of a collector restart. |
```
Comment (dmitryax) [inline receiver/filelogreceiver/README.md:32]: Should storage be marked as a required field?

Comment (djaglowski) [reply receiver/filelogreceiver/README.md:32]: It is not required. If not provided, file checkpoints will be managed in memory only.

Comment (dmitryax) [reply receiver/filelogreceiver/README.md:32]: It would be great if we can put add this comment to clarify

Comment (djaglowski) [reply receiver/filelogreceiver/README.md]: I've added some language about this

**Review by dmitryax — APPROVED** [review]

File: receiver/filelogreceiver/README.md (hunk 3 above)
Comment (dmitryax) [reply receiver/filelogreceiver/README.md]: Thanks!
